cds-snc/forms-terraform
# chore: Replace rds data client in reliability and nagware lambdas

# Summary | Résumé
This PR replaces the aws lib `client-rds-data` with the `Postgres.js` package due to the limitation of returning a 1MB maximum payload size.

`Postgres.js` was chosen over `node-postgres` due to it's simplicity and ease of use.  The ability to interpolate into the statement directly instead of using SQL parameters greatly simplifies the code structure.

# Test instructions | Instructions pour tester la modification
Setup:  Build Localstack Env & start GCForms app on `main` branch
- [ ] Test for Reliability Queue Lambda
1. Create a form a submit responses through the app
=> Verify that responses have been created and exist in the vault. 
=> Verify Lambda logs using LocalStack web interface to ensure there were no errors

- [ ] Test for Nagware Lambda 
1. Setup by executing a successful test above
2. Open NoSQL workbench and connect to the LocalStack DynamoDB service
3. Modify a response entry's `createdAt` to `1` to ensure it is caught as an old response.
4. run `awslocal lambda invoke --function-name nagware output.txt`
=> Verify that the responses have been deleted if the form was in draft
=> Verify Lambda logs using LocalStack web interface to ensure form was retrieved and tagged as needed to be nagged.
